specs/jsonschema-core.md

```diff
@@ -415,20 +415,43 @@ keywords MUST NOT begin with this prefix.
 Implementations MUST refuse to evaluate schemas which contain keywords which
 they do not know how to process or explicitly choose not to process.
 
-## Keyword Behaviors
+## Keyword Behaviors {#keyword-behaviors}
 
-JSON Schema keywords fall into several general behavior categories. Assertions
-validate that an instance satisfies constraints, producing a boolean result.
-Annotations attach information that applications may use in any way they see
-fit. Applicators apply subschemas to parts of the instance and combine their
-results.
+JSON Schema keywords may exhibit one or more behaviors. This specification
+defines three such behaviors[^7]:
+
+- Assertions validate that an instance satisfies constraints, producing a
+  boolean result: `true` if the constraints are satisfied; `false` otherwise.
+- Annotations attach information that applications may use in any way they see
+  fit.
+- Applicators apply subschemas to parts of the instance and combine their
+  results.
+
+[^7]: This specification also defines several operational directive keywords,
+such as `$id` and `$schema`. As such, these keywords do not exhibit these
+behaviors. However, it is recommended that extensions avoid defining additional
+directive keywords as they could interfere with schema processing and produce
+unexpected or undesirable results.
```

Comment:
I don't understand what you mean by "interfering with schema processing".

The way I see it, people should avoid creating these kinds of keywords because it's unlikely that an implementation will be able to support them using typical extension mechanisms. At least my implementation wouldn't.

Reply:
I've removed this sentence and expanded on the idea of a 'directive' keyword to explain why they're different.